### ISSUE TITLE
Refactor the server side to handle the new chart format

### ISSUE THREAD

**Issue by jackgr** [kubernetes/helm#367]
As noted in discussion on the Slack channel by @sparkprime and others, the server side components have not yet been refactored to work with the new chart format. 

Here are the things that need to be done:
- [x] Define a type that represents a chart.
- [x] Create expansion request that contains a single resource and a chart object, not a set of import files.
- [x] Refactor the registry package, renaming to repo, adding methods to resolve long and short chart references, and return chart objects populated with the contents of the referenced charts.
- [x] Replace TypeResolver with calls to repo provider to get charts.
- [x] Make ExpandTemplate examine the accumulated set of resources at each iteration, resolve chart references, and call the appropriate expander for each resource (i.e., may not be ExpandyBird).
- [x] Pass one resource at a time in the common.Template format to the expander, and revise the ExpandyBird service wrapper to unpack the new format.
- [x] Accept expansion results in the current ExpansionResult format and build the Layout, using the common.Layout returned by the expander, if any.
- [ ] Write a design document that describes refactoring.

**Comment by jackgr** [comment]
Depends on #400 to define the repo and repo provider interfaces.

**Comment by jackgr** [comment]
Closed by #462. Opening a new issue for the design document.
